Spring-server/src/main/java/ikpi63holding/postclient/data/user/User.java

```diff
@@ -1,74 +1,80 @@
 package ikpi63holding.postclient.data.user;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonManagedReference;
 import ikpi63holding.postclient.data.maibox.Mailbox;
-import java.util.List;
-import javax.persistence.Column;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.BinaryOperator;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
 import javax.persistence.Entity;
 import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
 import javax.persistence.Id;
 import javax.persistence.OneToMany;
-import javax.persistence.CascadeType;
+import javax.persistence.OrderBy;
 import lombok.Data;
+import org.hibernate.annotations.GenericGenerator;
 
 @Data
 @Entity
 public class User {
+
     @Id
-    @GeneratedValue(
-        strategy= GenerationType.AUTO, 
-        generator="native"
-    )
-    private Long id;
+    @GeneratedValue(generator = "uuid2")
+    @GenericGenerator(name = "uuid2", strategy = "org.hibernate.id.UUIDGenerator")
+    @Column(updatable = false, insertable = false, nullable = false)
+    private UUID id;
 
     @Column(nullable = false)
     private String username;
 
     @Column(nullable = false)
     private String password;
- 
-    private int active = 1;
 
+    private int active = 0;
+
+    @JsonIgnore
     private String roles = "";
 
+    @JsonIgnore
     private String permissions = "";
-  
-    @OneToMany(mappedBy = "user", cascade=CascadeType.ALL)
+
+    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL)
+    @OrderBy
     @JsonManagedReference
     private List<Mailbox> mailboxes;
 
     public void addMailbox(Mailbox mailbox) {
-        this.mailboxes.add(mailbox);
+        // TODO: later check for unique emailLogin here, and maybe throw otherwise
+        //  @Max
+        int maxId = mailboxes.stream().map(Mailbox::getId)
+                .reduce(BinaryOperator.maxBy(Integer::compareTo)).orElse(0);
+        int newId = maxId + 1;
```

Comment:
magic numbers

Reply:
Increment-operation will resolve it

Reply:
Is incrementing by one really considered so magic and required some documentation or variable? Postfix increment will modify original value for no real reason, or make additional unnecessary line.

Reply:
If zero is considered some magic, it is default maxId and I cam move it to static final variable, if this breaks the magic.

Reply:
> If zero is considered to be somewhat magical, it is default maxId and I can move it to static final variable, if this breaks the magic.

that make sense